x-pack/solutions/observability/plugins/observability/public/plugin.ts

```diff
@@ -231,8 +232,11 @@ export class Plugin
     const config = this.initContext.config.get();
     const kibanaVersion = this.initContext.env.packageInfo.version;
 
+    const isLegacyUptimeEnabled = coreSetup.uiSettings.get('observability:enableLegacyUptimeApp');
+
     this.observabilityRuleTypeRegistry = createObservabilityRuleTypeRegistry(
-      pluginsSetup.triggersActionsUi.ruleTypeRegistry
+      pluginsSetup.triggersActionsUi.ruleTypeRegistry,
+      isLegacyUptimeEnabled ? undefined : LEGACY_UPTIME_RULE_IDS
     );
```

Comment:
i don't think this makes sense, when calling from uptime plugin, the observability rules registry already takes into account this config and only register if conditions are met, adding a deny list here is duplicated. and this still won't solve the issue that's within stack management

Reply:
Do you have a suggested solution? This isn't really an issue with the fix, it's an issue with the way rule types get registered. Uptime alerts are defined in Kibana and can get registered from anywhere on the client, there's no in-built mechanism to disable that based on advanced config settings.

We could instead contribute a patch upstream in triggers actions that will respect the uptime advanced setting and skip registration for the rule from all clients, when legacy Uptime is disabled. Would you be ok with doing it that way instead?